pkg/clusteragent/custommetrics/provider.go

```diff
@@ -73,6 +73,10 @@ func (p *datadogProvider) ListAllExternalMetrics() []provider.ExternalMetricInfo
 	rawMetrics := p.hpaClient.ReadConfigMap()
 
 	for _, metric := range rawMetrics {
+		// Only metrics that exist in Datadog and available are eligible to be evaluated in the HPA process.
+		if !metric.Valid {
+			continue
```

Comment:
Perhaps a log here at the `Trace` level saying that this metric will be skipped.

Reply:
I think the log here: https://github.com/DataDog/datadog-agent/pull/1883/files/73125f99c98d2d80fc960bdd73de955aadb73eca..46e9c50c83abe9147b1bdcd6dffccf081e7a118a#diff-0fc3be6231b459e261c0d57460f1e200R213 and https://github.com/DataDog/datadog-agent/pull/1883/files/73125f99c98d2d80fc960bdd73de955aadb73eca..46e9c50c83abe9147b1bdcd6dffccf081e7a118a#diff-0fc3be6231b459e261c0d57460f1e200R219 along with the Validity bool are sufficient